airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -643,7 +660,16 @@ def read_records(
         while True:
             req = PreparedRequest()
             req.prepare_url(f"{job_full_url}/results", {"locator": salesforce_bulk_api_locator})
-            tmp_file, response_encoding, response_headers = self.download_data(url=req.url)
+            try:
+                tmp_file, response_encoding, response_headers = self.download_data(url=req.url)
+            except TRANSIENT_EXCEPTIONS as exception:
+                if call_count != 0:
+                    raise exception
```

Comment:
Need better logging to:
* mention we retry and for how many times
* mention we stop retrying